### PR TITLE
Remove improper &Arc<Blockstore> instances

### PR DESCRIPTION
**Pull request by steviez**
#### Summary of Changes
Update to either &Blockstore if the function just needs a ref, or Arc<Blockstore> if the function needs to hang onto a copy.

### CODE REVIEW

File: core/src/cache_block_meta_service.rs [hunk 1]
```diff
@@ -57,7 +57,7 @@ impl CacheBlockMetaService {
         Self { thread_hdl }
     }
 
-    fn cache_block_meta(bank: Arc<Bank>, blockstore: &Arc<Blockstore>) {
+    fn cache_block_meta(bank: Arc<Bank>, blockstore: &Blockstore) {
```
Comment (apfitzge) [inline core/src/cache_block_meta_service.rs:60]: we don't need an owned arc bank here either, and could use a `&Bank`; I think.

**Review by apfitzge — APPROVED** [review]

File: core/src/cache_block_meta_service.rs (hunk 1 above)
Comment (apfitzge) [reply core/src/cache_block_meta_service.rs:60]: can be done in a follow-up

File: core/src/sample_performance_service.rs [hunk 2]
```diff
@@ -21,10 +21,9 @@ pub struct SamplePerformanceService {
 impl SamplePerformanceService {
     pub fn new(
         bank_forks: &Arc<RwLock<BankForks>>,
-        blockstore: &Arc<Blockstore>,
+        blockstore: Arc<Blockstore>,
         exit: Arc<AtomicBool>,
     ) -> Self {
-        let blockstore = blockstore.clone();
         let bank_forks = bank_forks.clone();
```
Comment (apfitzge) [inline core/src/sample_performance_service.rs:27]: this could also be passed as owned

Comment (steviez) [reply core/src/sample_performance_service.rs:27]: Yeah, I noticed that one too; decided to keep this one limited to blockstores and one of us can take another pass on all the `&Arc<BankForks>` as I'm sure this isn't the only one

**Pull request merged**